rescript/tests/test_get_unite.py

```diff
@@ -0,0 +1,81 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2019-2023, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import pkg_resources
+import tempfile
+from qiime2.plugin.testing import TestPluginBase
+from rescript.get_unite import (
+    UNITE_DOIS,
+    _unite_get_url,
+    _unite_get_tgz,
+    _unite_get_artifacts,
+    get_unite_data,
+)
+
+from urllib.request import urlopen
+from urllib.error import HTTPError
+from unittest.mock import patch
+
+
+class TestGetUNITE(TestPluginBase):
+    package = "rescript.tests"
+
+    def setUp(self):
+        super().setUp()
+        self.unitefile = pkg_resources.resource_filename(
+            "rescript.tests", "data/unitefile.tgz"
+        )
+
+    # Requires internet access
+    def test_unite_get_url(self):
+        # for all combinations...
+        for v in UNITE_DOIS.keys():
+            for tg in UNITE_DOIS[v].keys():
+                for s in UNITE_DOIS[v][tg].keys():
+                    # ... try to get the URL
+                    try:
+                        url = _unite_get_url(v, tg, s)
+                        urlopen(url)
+                    except HTTPError:
```

Comment:
never reached in coverage, because I load everything from UNITE_DOIS

This makes me think I'm testing this function wrong. I'm trying to test my function, not their servers... right?

Reply:
The main point of this test, similar to [this](https://github.com/bokulich-lab/RESCRIPt/blob/946728467532f066c8bb750c3d8d1b8af02773b9/rescript/tests/test_get_gtdb.py#L47C9-L47C9), is to make sure the users machine can connect to these files. That is, does the use have internet access. So, this should always pass unless there is no internet access. Conceivably, it'll also raise an error if indeed the doi links changed too.

Reply:
What if I remove lines 44 and 45, then also remove the `try:` block so errors are uncaught and this fails the test?

Reply:
I think that'd work. You can also probably just add `self.assertTrue(True)` afterwards, you have an explicit "Test" being run.